### PR TITLE
[macOS] Add Colima tool

### PR DESCRIPTION
**Pull request by bogdan-damian-bgd**
# Description

Colima tool added to macOS

#### Related issue:

#6216 

## Virtual environments affected
- [ ] macOS 10.15
- [x] macOS 11
- [x] macOS 12

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/colima.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -e -o pipefail
```
Comment (al-cheb) [inline images/macos/provision/core/colima.sh:1]: Remove this file and use brew section in toolset file - https://github.com/actions/runner-images/blob/main/images/macos/toolsets/toolset-12.json

File: images/macos/templates/macOS-10.15.json [hunk 2]
```diff
@@ -203,7 +203,8 @@
                 "./provision/core/pypy.sh",
                 "./provision/core/pipx-packages.sh",
                 "./provision/core/bicep.sh",
-                "./provision/core/graalvm.sh"
+                "./provision/core/graalvm.sh",
+                "./provision/core/colima.sh"
```
[the remark this answers is not in the document]
Comment (bogdan-damian-bgd) [reply images/macos/templates/macOS-10.15.json]: Yes, I removed it for the 10.15 version

File: images/macos/provision/core/colima.sh (hunk 1 above)
Comment (bogdan-damian-bgd) [reply images/macos/provision/core/colima.sh]: I switched to toolset json

**Review by al-cheb — APPROVED** [review]

File: images/macos/software-report/SoftwareReport.Generator.ps1 [hunk 3]
```diff
@@ -182,7 +182,8 @@ $toolsList += @(
     (Get-GHCVersion),
     (Get-CabalVersion),
     (Get-StackVersion),
-    (Get-SwiftFormatVersion)
+    (Get-SwiftFormatVersion),
+    (Get-ColimaVersion)
```
Comment (miketimofeev) [inline images/macos/software-report/SoftwareReport.Generator.ps1:186]: @bogdan-damian-bgd should it be in the runtime section instead as Colima - container **runtimes** on macOS (and Linux) with minimal setup?

Comment (bogdan-damian-bgd) [reply images/macos/software-report/SoftwareReport.Generator.ps1:186]: Yes, directly from a naming point of view it should be in 'Language and Runtimes'.

On the other hand, if we consider it as some kind of replacement for Docker (which is also labeled as a container **runtime**) and try to be consistent with the location of Docker within the ReadMe sections for Windows and Linux, then it also makes some sense to stay in this section as well?

Comment (miketimofeev) [reply images/macos/software-report/SoftwareReport.Generator.ps1:186]: Yup, if we'd like to be consistent then it's fine

**Review by igorboskovic3 — APPROVED** [review]

**Pull request merged**